prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -235,6 +228,29 @@ def _storage_root_uri(cls, system_id: str, storage_name: str) -> str:
         """
         return cls.systems_root_uri + f"{system_id}/{storage_name}/"
 
+    def _find_storage_controller_data(self, storage_data: Dict) -> List[Dict]:
+        """Return storage controller list.
+
+        Returns:
+            storage_controller_list: list of storage controllers for
+            requested storage id.
+
+        """
+        storage_controllers_list: List[Dict]
+        if "StorageControllers" in storage_data:
+            storage_controllers_list = storage_data["StorageControllers"]
+        # ILO version is 2.99 might have "Controllers" key instead of "StorageControllers"
```

Comment:
**question (non-blocking):** 

Is that possible we know the version first and choose the specific function to parse the API response, instead of make this function to support all the possible versions? Or you think it's unnecessary here?

Reply:
Feel free to resolve this one.